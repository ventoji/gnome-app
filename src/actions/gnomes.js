
export const addGnomes = (gnomes) => ({
    type: 'ADD_GNOMES',
    gnomes
  });