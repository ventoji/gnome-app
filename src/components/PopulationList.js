import React, {useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import SvgIcon from './SvgIcon';
import pathApp from '../urlApp/index';

const PopulationList = (props) => {

    const [url, setUrl] = useState('https://raw.githubusercontent.com/rrafols/mobile_test/master/data.json');
    const [isLoading, setIsLoading] = useState(false);

    const fetchGnomes = () => {
        setIsLoading(true);
        fetch(url)
           .then( result => result.json())
           .then( data => {
             const {Brastlewark} = data;
            props.addGnomes(Brastlewark);
            setIsLoading(false);
           })
           .catch(error => console.log(error));
      };

      useEffect(() => {
        fetchGnomes();

        return () => {
         // console.log("cleanup");
        };
      },[]); 

    const showLoading = () => (
         isLoading ? <div className="loader">
         <img className="loader__image" src={`${pathApp}img/Spinner-1s-200px.gif`}/>
     </div> : ''
      );

    const showGnomesFiletered = () => {
    
      return props.gnomes.map((gnome, i) =>  <li className="population__list-item" key={i}>
    
      <p> {gnome.name }     </p>  
      <span> <Link className="population__list-link"  to={`${pathApp}gnome/${gnome.id}`} onClick={ () => {
        props.addGnomeItem(gnome);
  
    }}>   
    <SvgIcon svgClass="icon--gnome" svgName="img/sprite.svg#icon-eye" />
      </Link>  </span> 
  </li> );
    };


    return (
     
      <section className="population-home">

        <div className="population">
        
        {showLoading()}

        
        <div className="population-list">
   
        <ul className="population__gnomes">
      
       
        {isLoading ? '' : showGnomesFiletered()}

        </ul>
     
        </div>
      
        </div>
        </section>
    );
};

export default PopulationList;