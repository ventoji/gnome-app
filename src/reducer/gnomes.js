export default (state=[], action) => {

    switch (action.type){

        case 'ADD_GNOMES':
            return  action.gnomes;

        default:
            return state;

    }
};