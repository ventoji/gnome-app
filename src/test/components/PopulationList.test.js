import  PopulationList  from '../../components/PopulationList';
import { shallow } from 'enzyme';
import React from 'react';
import gnomes from '../fixtures/gnomes';


let wrapper;

describe(`The Gnome Filter Component`, () => {

    describe(`The display Element`, () => {

        beforeEach(() => {
            wrapper = shallow(<PopulationList gnomes={gnomes} />);
          });

          it(`Should render component properly`, () => {
            expect(wrapper).toMatchSnapshot();
        }); 

    });
});