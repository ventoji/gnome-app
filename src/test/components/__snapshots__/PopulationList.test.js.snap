// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Gnome Filter Component The display Element Should render component properly 1`] = `
<section
  className="population-home"
>
  <div
    className="population"
  >
    <div
      className="population-list"
    >
      <ul
        className="population__gnomes"
      >
        <li
          className="population__list-item"
          key="0"
        >
          <p>
             
            Tobus Quickwhistle
                 
          </p>
          <span>
             
            <Link
              className="population__list-link"
              onClick={[Function]}
              to="/gnome/0"
            >
              <SvgIcon
                svgClass="icon--gnome"
                svgName="img/sprite.svg#icon-eye"
              />
            </Link>
              
          </span>
        </li>
        <li
          className="population__list-item"
          key="1"
        >
          <p>
             
            Fizkin Voidbuster
                 
          </p>
          <span>
             
            <Link
              className="population__list-link"
              onClick={[Function]}
              to="/gnome/1"
            >
              <SvgIcon
                svgClass="icon--gnome"
                svgName="img/sprite.svg#icon-eye"
              />
            </Link>
              
          </span>
        </li>
        <li
          className="population__list-item"
          key="2"
        >
          <p>
             
            Malbin Chromerocket
                 
          </p>
          <span>
             
            <Link
              className="population__list-link"
              onClick={[Function]}
              to="/gnome/2"
            >
              <SvgIcon
                svgClass="icon--gnome"
                svgName="img/sprite.svg#icon-eye"
              />
            </Link>
              
          </span>
        </li>
      </ul>
    </div>
  </div>
</section>
`;
